### PR TITLE
fix: added initial implementation for supporting internal function definitions

### PR DESCRIPTION
**Pull request by arteevraina**
Towards: #2006

### CODE REVIEW

File: integration_tests/func_internal_def_01.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+def main():
+    x: i32
+    x = (2+3)*5
+    print(x)
+
+    def bar():
+        print("bar")
```
Comment (certik) [inline integration_tests/func_internal_def_01.py:7]: Add here:
```python
assert x == 25
```

**Comment by certik** [comment]
We are getting some error messages at the CI:
```
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0x9a01d4
  Binary file "/lib/x86_64-linux-gnu/libc.so.6", local address: 0x29e3f
  Binary file "/lib/x86_64-linux-gnu/libc.so.6", local address: 0x29d8f
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0x9afc09
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0x9a71c7
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0xa7699d
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0xa76344
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0xaac25c
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0xadf5ed
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0xa8aafd
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0xaaef5f
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0xadf5ed
  Binary file "/home/runner/work/lpython/lpython/src/bin/lpython", local address: 0xa8ad19
LCompilersException: visit_Pass() not implemented
make[2]: *** [CMakeFiles/const_03_FAST.dir/build.make:74: const_03_FAST.o] Error 1
```
Not sure where they care coming from, but we need to fix it.

**Comment by arteevraina** [comment]
I have removed it to test https://github.com/lcompilers/lpython/pull/2119/commits/551997b5cd02df7ecb53a0e804fdf2450077d77b. Let's see if ci works now.

**Comment by certik** [comment]
The failure is:
```
make[2]: *** [CMakeFiles/func_internal_def_01_FAST.dir/build.make:74: func_internal_def_01_FAST.o] Segmentation fault (core dumped)
make[1]: *** [CMakeFiles/Makefile2:5822: CMakeFiles/func_internal_def_01_FAST.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -669,6 +669,7 @@ RUN(NAME func_static_01 LABELS cpython llvm c wasm)
 RUN(NAME func_static_02 LABELS cpython llvm c wasm)
 RUN(NAME func_dep_03    LABELS cpython llvm c)
 RUN(NAME func_dep_04    LABELS cpython llvm c)
+RUN(NAME func_internal_def_01 LABELS cpython llvm)
```
Comment (certik) [inline integration_tests/CMakeLists.txt:672]:
```suggestion
RUN(NAME func_internal_def_01 LABELS cpython llvm NOFAST)
```

File: ci/test.xsh [hunk 3]
```diff
@@ -22,8 +22,8 @@ else:
     src/bin/lpython examples/expr2.py
     src/bin/lpython --backend=c examples/expr2.py
     cd integration_tests
-    python run_tests.py -j16 -b llvm cpython c wasm
-    python run_tests.py -j16 -b llvm cpython c wasm -f
+    python run_tests.py -j16 -b llvm cpython wasm
+    python run_tests.py -j16 -b llvm cpython wasm -f
```
Comment (certik) [inline ci/test.xsh:26]:
```suggestion
    python run_tests.py -j16 -b llvm cpython c wasm
    python run_tests.py -j16 -b llvm cpython c wasm -f
```

**Comment by arteevraina** [comment]
Hi @certik ,
Sorry, tried updating this branch with base branch and forced pushed without pulling your changes. Sorry about that.
But, I have added those changes in my new commits and marked this PR as ready for review.

**Pull request merged**